src/databricks/labs/ucx/install.py

```diff
@@ -676,17 +675,39 @@ def _get_workspace(workspace_id: int, ids_to_workspace: dict[int, Workspace]) ->
             msg = "Current workspace is not known, Please run as account-admin: databricks labs ucx sync-workspace-info"
             raise KeyError(msg) from None
 
-    def _get_workspace_info(self, current_workspace_id: int):
+    def _get_workspace_ids(self):
         account_client = self._get_safe_account_client()
-        workspace = account_client.workspaces.get(current_workspace_id)
-        current_workspace_client = account_client.get_workspace_client(workspace)
-        installation = Installation.current(current_workspace_client, self.product_info.product_name())
-        workspace_info = WorkspaceInfo(installation, current_workspace_client)
-        return workspace_info.load_workspace_info()
+        workspaces = account_client.workspaces.list()
+        ids_to_workspace = {}
+        for workspace in workspaces:
+            ids_to_workspace[workspace.workspace_id] = workspace
+        return ids_to_workspace
+
+    def get_workspace_contexts(self, collection_workspace_id: int) -> list[WorkspaceContext]:
+        workspace_contexts = []
+        account_client = self._get_safe_account_client()
+        acct_ctx = AccountContext(account_client)
+        collection_workspace = account_client.workspaces.get(collection_workspace_id)
+        if not acct_ctx.account_workspaces.can_administer(collection_workspace):
+            logger.error(f"User is not workspace admin of collection workspace {collection_workspace_id}")
+            return []
+        installer = self._get_installer(collection_workspace)
+        if installer.config.installed_workspace_ids is None:
+            logger.error(f"No collection info found in the workspace {collection_workspace_id}")
+            return []
+        for workspace_id in installer.config.installed_workspace_ids:
+            workspace = account_client.workspaces.get(workspace_id)
+            if not acct_ctx.account_workspaces.can_administer(workspace):
```

Comment:
shouldn't this move/be integrated with `ctx.account_workspaces.get_accessible_workspaces()`? e.g. you get accessible workspaces once and you assume you can administer those.

Reply:
have used account_workspaces.get_accessible_workspace() to get list of accessible workspace and using that to validate if user can join a collection